src/mutation-observer.js

```diff
@@ -63,24 +63,38 @@ const clearNode = node => {
 	}
 }
 
+const isTramOneComponent = node => {
+	if (node[TRAM_TAG_NEW_EFFECTS] !== undefined) {
+		return NodeFilter.FILTER_ACCEPT
+	}
+
+	return NodeFilter.FILTER_SKIP
+}
+
 // function to get the children (as a list) of the node passed in
-const childrenNodes = node => {
-	const children = node.querySelectorAll ? node.querySelectorAll('*') : []
-	return [...children]
+// this only needs to query tram-one components, so we can use the attribute `tram`
+const childrenComponents = node => {
+	const componentWalker = document.createTreeWalker(node, NodeFilter.SHOW_ELEMENT, isTramOneComponent)
+	const children = []
+	while (componentWalker.nextNode()) {
+		children.push(componentWalker.currentNode)
+	}
+
+	return children
```

Comment:
rather than using `querySelectorAll` (which forces reflow), we use a `TreeWalker`! This `TreeWalker` specifically looks for nodes that pass the `isTramOneComponent` check (which in turn, checks if the node has `TRAM_TAG_NEW_EFFECTS` - which will be some array, even empty)